### PR TITLE
Simplify FindIWADFile and CheckIWAD

### LINKED ISSUE #498: [Suggestion] Relpace FindIWADFile and CheckIWAD
I suggest replacing `FindIWADFile` and `CheckIWAD` with the Crispy/Choco counterparts from `d_iwad.c`. These functions are required for the multiplayer launcher but are very difficult to read/maintain and buggy. This is my Windows output for the wrong -iwad option:
```
Looking in .
Looking in .
Looking in C:\Users\User
Looking in \doom2
Looking in \plutonia
Looking in \tnt
Looking in \doom_se
Looking in \doom
Looking in \dooms
Looking in \doomsw
Looking in C:\Users\User
IWAD not found
```
Obviously there is a bug in `FindIWADFile`, but it's much easier to write a new function than to modify it. These functions should be trivial.

### PR DESCRIPTION
**Pull request by rfomin**
Fixes #498

### CODE REVIEW

File: Source/d_main.c [hunk 1]
```diff
@@ -743,281 +747,99 @@ static void PrepareAutoloadPaths (void)
...
-        return false;
-      return true;                //otherwise return file found, w/ .wad added
+        if (!strcasecmp(name, standard_iwads[i].name))
+        {
+            *gmode = standard_iwads[i].mode;
+            *gmission = standard_iwads[i].mission;
+
+            if (*gmode == commercial)
+            {
+                *hassec = CountMaps(iwadname) < 31 ? false : true;
```
Comment (rfomin) [inline Source/d_main.c:799]: I think it's enough to determinate IWAD by name. Do we need check for wolf levels here?

Comment (fabiangreffrath) [reply Source/d_main.c:799]: We need this to set the `haswolflevels` variable.

Comment (rfomin) [reply Source/d_main.c:799]: Yes, but we can remove `haswolflevels`, and fix it in Choco/Crispy way.

Comment (fabiangreffrath) [reply Source/d_main.c:799]: We could check for the existence of the two map lumps once the IWAD is loaded, if that's what you mean.

Comment (fabiangreffrath) [reply Source/d_main.c:799]: Yes, please.

**Comment by fabiangreffrath** [comment]
So, with this approach we don't do any content checking at all anymore?

**Comment by rfomin** [comment]
> So, with this approach we don't do any content checking at all anymore?

Yes, and I think it's okay. Choco/Crispy don't do that either, right?

**Comment by fabiangreffrath** [comment]
Yeah, should be alright. No idea why Boom did this special pony thing.

**Comment by fabiangreffrath** [comment]
This allows for command lines like `woof -iwad dOOm2`:
```patch
--- a/Source/d_main.c
+++ b/Source/d_main.c
@@ -781,11 +781,15 @@ char *FindIWADFile(void)
 
         char *iwadfile = myargv[iwadparm + 1];
 
-        result = D_FindWADByName(iwadfile);
+        char *file = (malloc)(strlen(iwadfile) + 5);
+        AddDefaultExtension(strcpy(file, iwadfile), ".wad");
+
+        result = D_FindWADByName(file);
+        (free)(file);
 
         if (result == NULL)
         {
-            I_Error("IWAD file '%s' not found!", iwadfile);
+            I_Error("IWAD file '%s' not found!", file);
         }
     }
     else
```

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**